src/tests/ftest/launch.py

```diff
@@ -52,2294 +68,2973 @@
+
+        # Create a temporary remote directory
+        command = f"mkdir -p {tmp_copy_dir}"
+        if not run_remote(logger, hosts, command).passed:
+            message = f"Error creating temporary remote copy directory {tmp_copy_dir}"
+            self._fail_test(self.result.tests[-1], "Process", message)
+            return 16
+
+        # Move all the source files matching the pattern into the temporary remote directory
+        other = f"-print0 | xargs -0 -r0 -I '{{}}' {sudo_command}mv '{{}}' {tmp_copy_dir}/"
+        if not run_remote(logger, hosts, find_command(source, pattern, depth, other)).passed:
+            message = f"Error moving files to temporary remote copy directory {tmp_copy_dir}"
+            self._fail_test(self.result.tests[-1], "Process", message)
+            return 16
+
+        # Clush -rcopy the temporary remote directory to this host
+        command = ["clush", "-w", str(hosts), "-v", "--rcopy", tmp_copy_dir, "--dest", rcopy_dest]
+        return_code = 0
+        try:
+            run_local(logger, command, check=True, timeout=timeout)
 
-    """
-    result = {"status": 0}
-    status_keys = ["reset-failed", "stop", "disable"]
-    mapping = {"stop": "active", "disable": "enabled", "reset-failed": "failed"}
-    check_hosts = NodeSet(hosts)
-    loop = 1
-    # Reduce 'max_loops' to 2 once https://jira.hpdd.intel.com/browse/DAOS-7809
-    # has been resolved
-    max_loops = 3
-    while check_hosts:
-        # Check the status of the service on each host
-        result = get_service_status(check_hosts, service)
-        check_hosts = NodeSet()
-        for key in status_keys:
-            if result[key]:
-                if loop == max_loops:
-                    # Exit the while loop if the service is still running
-                    log(" - Error {} still {} on {}".format(service, mapping[key], result[key]))
-                    result["status"] = 512
-                else:
-                    # Issue the appropriate systemctl command to remedy the
-                    # detected state, e.g. 'stop' for 'active'.
-                    command = "sudo systemctl {} {}".format(key, service)
-                    get_remote_output(result[key], command)
+        except RunException:
+            message = f"Error copying remote files to {destination}"
+            self._fail_test(self.result.tests[-1], "Process", message, sys.exc_info())
+            return_code = 16
 
-                    # Run the status check again on this group of hosts
-                    check_hosts.add(result[key])
-        loop += 1
-    return result["status"]
+        finally:
+            # Remove the temporary remote directory on each host
+            command = f"{sudo_command}rm -fr {tmp_copy_dir}"
+            if not run_remote(logger, hosts, command).passed:
+                message = f"Error removing temporary remote copy directory {tmp_copy_dir}"
+                self._fail_test(self.result.tests[-1], "Process", message)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'fail_reason' in staticmethod call